### ISSUE TITLE
[Bug]: `create_from_BPX` method incompatible with "lumped" thermal model

### ISSUE THREAD

**Issue by darryl-ad** [pybamm-team/PyBaMM#2860]
### PyBaMM Version

23.1

### Python Version

3.9.12

### Describe the bug

The parameter set created using the `create_from_BPX` method is not compatible with the "lumped" thermal model.

The following issues exist:
1. The "lumped" model calls the current collector variables, which `create_from_BPX` does not generate. These include density, heat capacity, thermal conductivity, thickness, and conductivity.

2. The specific heat capacities generated by `create_from_BPX` are of units "[J.K-1.kg-1]" instead of the "[J.kg-1.K-1]" expected by PyBaMM. This applies to the negative electrode, positive electrode, and separator specific heat capacities.

### Steps to Reproduce

`params = pybamm.ParameterValues.create_from_bpx("exampleBPX.json")`

`sim = pybamm.Simulation(
    pybamm.lithium_ion.DFN(options={"thermal":"lumped", "cell geometry": "arbitrary"}),
    parameter_values=params
)`

`sol = sim.solve()`

### Relevant log output

```shell
{
	"name": "KeyError",
	"message": "\"'Negative current collector density [kg.m-3]' not found. Best matches are ['Negative electrode density [kg.m-3]', 'Positive electrode density [kg.m-3]', 'Separator density [kg.m-3]']\"",
	....
```

**Comment by rtimms** [comment]
Thanks for pointing this out! Currently we assign the lumped value to each component but we miss the current collectors. Should be an easy fix by adding the current collectors domains to [this line](https://github.com/pybamm-team/PyBaMM/blob/0c1282675148e7ec85a67ff8c10293497e67ab4c/pybamm/parameters/bpx.py#L117).  Do you think you could have a go?

**Comment by darryl-ad** [comment]
Thanks @rtimms, I'm happy to give this a go. However I don't think your quick fix will cover all the issues listed, just the current collector density, heat capacity, and thermal conductivity. Let me know what you think of the following:

- The current collector thickness and conductivity parameters are still required by the "lumped" thermal model, but aren't part of BPX. What would be the repurcussions of zeroing these parameters out?
- I'll rearrange the units in [this line](https://github.com/pybamm-team/PyBaMM/blob/0c1282675148e7ec85a67ff8c10293497e67ab4c/pybamm/parameters/bpx.py#L113) to address point 2.

**Comment by rtimms** [comment]
- The current collector thicknesses will be arbitrary for this setup - they’re only used to compute a weighted mean of the thermal properties, but since they’re hard-coded to be the same the end result won’t care what the thickness are. So, for a quick fix, either zero out or pick some nominal value. 
- Thanks. 

The better fix for this would be to use the lumped properties directly if you choose `{“thermal”: “lumped”, “cell geometry”: “arbitrary”}` and compute them by component if you choose `{“thermal”: “lumped”, “cell geometry”: “pouch”}`. Similar to how we do surface area for cooling and cell volume [here](https://github.com/pybamm-team/PyBaMM/blob/cd2d1c8c7c09cfc16c1c885c1e6612628926e38c/pybamm/models/submodels/thermal/lumped.py#L65). This calculation could probably be moved to `LithiumIonParameters`.

**Comment by rtimms** [comment]
@darryl-ad any progress on this?

**Comment by darryl-ad** [comment]
@rtimms sorry I've been on holiday. I'll get to it soon so you can include it in the 23.5 release.

I just need to get up to speed on your [contribution guidelines](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md) and dev install instructions

**Comment by RuiheLi** [comment]
Hi~ @darryl-ad, I will start working on it today.

**Comment by darryl-ad** [comment]
@RuiheLi I'm set up now so I'm happy to do it :)

**Comment by rtimms** [comment]
Fixed by #2987
